### PR TITLE
Presto on Spark small refactorings and bug fixes

### PR DESCRIPTION
**Pull request by arhimondr**
Required to prepare Presto on Spark for running AbstractTestQueries

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/src/main/java/com/facebook/presto/spark/node/PrestoSparkInternalNodeManager.java [hunk 1]
```diff
@@ -38,30 +38,34 @@
...
     {
-        throw new UnsupportedOperationException();
+        return ImmutableSet.of(getCurrentNode());
     }
 
     @Override
     public AllNodes getAllNodes()
     {
-        throw new UnsupportedOperationException();
+        return new AllNodes(
```
Comment (shixuan-fan) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/node/PrestoSparkInternalNodeManager.java:60]: nit: maybe this should be a static field instead?

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkInjectorFactory.java [hunk 2]
```diff
@@ -44,13 +45,27 @@
...
+            List<Module> additionalModules)
+    {
+        this(configProperties, catalogProperties, additionalModules, Optional.empty());
+    }
+
+    public PrestoSparkInjectorFactory(
+            Map<String, String> configProperties,
+            Map<String, Map<String, String>> catalogProperties,
+            List<Module> additionalModules,
+            Optional<Module> accessControlModuleOverride)
```
Comment (shixuan-fan) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkInjectorFactory.java:62]: I'm not particularly sure if we should have two separate constructors, but I don't have strong opinion on this.

File: presto-spark-base/src/main/java/com/facebook/presto/spark/node/PrestoSparkInternalNodeManager.java (hunk 1 above)
Comment (arhimondr) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/node/PrestoSparkInternalNodeManager.java:60]: It is not going to be called on production anyway. It's pretty much a hack just to make TPCH / System connectors happy. Hopefully we will refactor `presto-main` a little bit and decouple it from the node management.

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkInjectorFactory.java (hunk 2 above)
Comment (arhimondr) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkInjectorFactory.java:62]: Just wanted to keep the main non testing call site clean. Having an extra optional might be confusing.

File: presto-spark-base/src/main/java/com/facebook/presto/spark/node/PrestoSparkInternalNodeManager.java (hunk 1 above)
Comment (arhimondr) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/node/PrestoSparkInternalNodeManager.java]: Extracted static constants

**Pull request merged**